interpreter/exec/eval.ml

```diff
@@ -78,7 +78,10 @@ and admin_instr' =
 and ctxt = code -> code
 
 type cont = int * ctxt  (* TODO: represent type properly *)
-type ref_ += ContRef of cont option ref
+type cont_status = Alive of cont
```

Comment:
AFAICS, there is no semantic difference between Taken and Destroyed, so why introduce this type?

Reply:
There is a conceptual difference, though. I suppose it would only manifest if we were to provide an instruction for continuation copying. I can revert the commit / stash it for later.

Reply:
Hm, isn't copying orthogonal as well? Wouldn't it simply map Alive to Alive? How would it observably distinguish the other two?

Reply:
Thinking about this some more, having `resume_throw` behave differently from `resume` with a special Destroyed state doesn't make sense to me in the first place. Either continuations are linear (with or without explicit copy), then both instructions should destroy it, or they are not, then neither should. In particular, given
```
(func $foo (param $f (ref $proc)) (result (ref $cont))
  (block $l (result (ref $cont))
    ... (resume (event $evt $l) (cont.new $f)) ...
  )
)
```
this program
```
(func $f ... (suspend $evt) ...)
(func $g ... (resume_throw $exn (arg)* (call $foo (ref.func $f))) ... )
```
should be equivalent to this
```
(func $f' ... (suspend $evt) (throw $exn) ...)
(func $g' ... (resume (arg)* (call $foo (ref.func $f'))) ... )
```